### PR TITLE
Feature/nh/solver manager

### PR DESCRIPTION
**Pull request by TotallyRealHolm**
I made a basic version of the solution manager, it however  does not make use of the correct solver image yet, so that will have to be changed in the future

### CODE REVIEW

File: SolverManager/src/test/java/com/oops/solvermanager/AppTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import org.junit.Test;
+
+/**
+ * Unit test for simple App.
+ */
+public class AppTest 
+{
+    /**
+     * Rigorous Test :-)
```
Comment (Maje419) [inline SolverManager/src/test/java/com/oops/solvermanager/AppTest.java:13]: I like your rigor. 10/10 xD

File: SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    private void createSolverJobs(ProblemRequest newProblem) {
+        KubernetesClient api = makeKubernetesClient();
+        for (SolverBody solver : newProblem.getSolversToUse()) {
+            List<String> command = new LinkedList<>();
+            command.add("perl");
+            command.add("-Mbignum=bpi");
+            command.add("-wle");
+            command.add("print bpi(2000)");
```
Comment (Maje419) [inline SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java:84]: I get that these are here so you can test it, but they should probably be changed to add the actual commands that the solver takes before pushing. Keep this stored on you local tho.

```suggestion
            command.add("python3");
            command.add("Solver.py");
            command.add(solver.getSolverName());
            command.add(solver.numberVCPU());
            command.add(newProblem.getProblemId());
            command.add(newProblem.getDataId());
```

File: SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    .withName(newProblem.getProblemID().toLowerCase() + solver.getSolverName().toLowerCase())
+                    .withLabels(labels)
+                    .endMetadata()
+                    .withNewSpec()
+                    .withTtlSecondsAfterFinished(30)
+                    .withNewTemplate()
+                    .withNewSpec()
+                    .addNewContainer()
+                    .withName("pi")
+                    .withImage("perl:5.34.0")
```
Comment (Maje419) [inline SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java:101]: Same as above, use the correct image :D

File: SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                    .withBackoffLimit(4)
+                    .endSpec()
+                    .build();
+            api.batch().v1().jobs().inNamespace("default").resource(job).create();
+        }
+    }
+
+    private KubernetesClient makeKubernetesClient() {
+        return new KubernetesClientBuilder().build();
+
```
Comment (Maje419) [inline SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java:116]:
```suggestion
```

File: SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
+package com.oops.solvermanager;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
```
Comment (Maje419) [inline SolverManager/src/main/java/com/oops/solvermanager/SolverManagerController.java:10]: Get rid of any unused imports :100:

**Comment by TotallyRealHolm** [comment]
Resolved the comments written in the pull request

**Review by Maje419 — APPROVED** [review]
Nice work!

**Pull request merged**